libusb/core.c

```diff
@@ -2463,8 +2463,12 @@ int API_EXPORTED libusb_init_context(libusb_context **ctx, const struct libusb_i
 	if (getenv("LIBUSB_DEBUG")) {
 		_ctx->debug = get_env_debug_level();
 		_ctx->debug_fixed = 1;
-	} else if (default_context_options[LIBUSB_OPTION_LOG_LEVEL].is_set) {
-		_ctx->debug = default_context_options[LIBUSB_OPTION_LOG_LEVEL].arg.ival;
+	} else {
+		usbi_mutex_static_lock(&default_context_lock);
```

Comment:
This lock causes a (self) deadlock because of the same lock being previously acquired at line 2438.

Reply:
@sonatique thanks for reviewing! Yes, you are quite right. Not sure what I was thinking here... will amend this patch.

Reply:
And `-Wthread-safety` even warns about this. Doh.

I fear I accidentally committed some WIP. I have too many different libusb branches going simultaneously...

Reply:
OK, I've just dropped that 1 commit.